src/win/pipe.c

```diff
@@ -357,7 +357,12 @@ int uv_pipe(uv_file fds[2], int read_flags, int write_flags) {
   /* TODO: better source of local randomness than &fds? */
   read_flags |= UV_READABLE_PIPE;
   write_flags |= UV_WRITABLE_PIPE;
-  err = uv__create_pipe_pair(&readh, &writeh, read_flags, write_flags, 0, (char*) &fds[0]);
+  err = uv__create_pipe_pair(&readh,
+                             &writeh,
+                             read_flags,
+                             write_flags,
+                             0,
+                             (unsigned long long)&fds[0]);
```

Comment:
```suggestion
                             (uintptr_t) &fds[0]);
```
nit: long long is not the size of a pointer

Reply:
But we are not trying to fit a pointer in an integer per se. We want a random number, to be used for generating a unique pipe name, which avoids collisions.

I think unsigned long long has enough bits here, it's unlikely we would have so many open pipes we would see collisions in practice. The (pre-existing) hack though is trying to use a heap pointer as a source of randomness. Granted, if pointer size were larger than long long, we would be discarding some bits, but do you really think this is a problem?

Reply:
That's right, but the HANDLE pointer needs to be cast first to an uintptr_t-sized object before up-or-downcasting the bytes (to make the compiler happy). We may want to consider using SystemFunction036 or `uv_random` eventually, since this is indeed a pretty ridiculous hack (which I say as the person who wrote it).

Reply:
While the standard does not say anything directly about pointer sizes wrt integer sizes, unsigned long long is the largest integer available in standard C.

So if for a given platform a pointer could round trip through a suitably large integer, this integer would necessarily be at most as large as ULL.

Reply:
Yes, that is fine, but the compiler won't like it if you cast directly to a different size

Reply:
I don't think a C compiler could realistically complain about a pointer cast to an unsigned int wider than necessary, as the amount of C code that predates uintptr_t is quite high.

But I did the change anyway considering it might be a style / clearer intent thing.

Reply:
I don't make the rules. It is not allowed by gcc:
```c
$ echo 'int main() { return (long long)&main; }' | gcc -x c -o /dev/null -Wall - -m32
<stdin>: In function ‘main’:
<stdin>:1:21: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
```

Reply:
I see, yeah that's a bad warning IMO, gladly it's only in Wall, and clang doesn't do it.